Improve documentation (#305)
eiffel-community/eiffel: eiffel-syntax-and-usage/event-design-guidelines.md

* __Key names in camelCase:__ All key names shall use camelCase.
* __Enumerations in CAPS_WITH_UNDERSCORE:__ All enumerated values shall use CAPS_WITH_UNDERSCORE.
* __Link types are nouns:__ Link types shall have names that fit the pattern "\<target event> is the \<source event>'s \<link type>", implying that they are nouns that describe the relationship between the source and the target event.

## Benefits and Implications

By following the design guidelines the Eiffel protocol provides a language that
gives producer and consumer a common way of understanding and describing the
occurrences in the system by defining a contract between them.

Following the guidlines stated, they adress the following issues:

> * __Dialects__: By using the protocol it gives the best possible conditions

Reviewer: "Dialects" might be hard to understand for a new reader. Maybe "Protocol variants - dialects for the Eiffel language"

Author: Changed to "Protocol variants"

  for a non-initiated event consumer to make sense of any given set of events.

> * __Inconsistency__: By providing clear explanations of each parameter for both

Reviewer 2: Again, "parameter" probably isn't the best word. Prefer "member".

Author: Yes, used "member" instead

  producer and consumer it minimizes the risk for both data divergence and
  inconsistency.
* __Coordination__: The Eiffel protocol minimizes coordination effort between
  producer and consumer by giving a well-defined specification by defining a

>   contract between the producer and the consumer.

Reviewer 2: This sentence doesn't read too well (two subclauses beginning with "by XXXing"). Something like this?

> The Eiffel protocol minimizes coordination effort between producers and consumers by defining a specification that the parties can use as a contract.

Author: Thanks, reworded it based on you suggestion

Reviewer 2: Better, but there are a couple of remaining nits. Suggestion:

The Eiffel protocol minimizes coordination efforts between  producers and consumers by providing a well-defined specification that the  parties can use as a contract.

Author: Fixed in [5de3f4b](https://github.com/eiffel-community/eiffel/pull/305/commits/5de3f4b54e8307affa3cf520065c8009c27434f7) ( and do hope I got it this time)

> * __Context__: Following the protocol gives each data a clear context to which it belongs

Reviewer 2: Missing full stop at the end of the sentence?

Author: Yes, fixed.

* __Hard dependencies__: By using the Eiffel protocol specification it reduces the
  risk for creating hard connections between producers and consumers. We want to
  enable adding new consumers without affecting the producers. This then leads to
  a decoupled system with separation of concerns than enables scalability and

>   maintainability.

Reviewer 2: While "hard dependency" is a real thing I'm not sure it's what you're describing here. Maybe "tight coupling" would be more apt? Also, s/risk for/risk of/ and s/than enables/that enables/.

Author: Good point, I changed it now to "Tight coupling" instead.

* __Key names in camelCase:__ All key names shall use camelCase.
* __Enumerations in CAPS_WITH_UNDERSCORE:__ All enumerated values shall use CAPS_WITH_UNDERSCORE.
* __Link types are nouns:__ Link types shall have names that fit the pattern "\<target event> is the \<source event>'s \<link type>", implying that they are nouns that describe the relationship between the source and the target event.

## Benefits and Implications

By following the design guidelines the Eiffel protocol provides a language that

> gives producer and consumer a common way of understanding and describing the

Reviewer 2: This is probably not a hard rule, but in this case I just think it sounds better with the plural form:

```suggestion
gives producers and consumers a common way of understanding and describing the
```

There are other similar cases, I won't make suggestions for all of them.

occurrences in the system by defining a contract between them.

> Following the guidlines stated, they adress the following issues:

Reviewer 2: ```suggestion
Following the guidelines stated, they address the following issues:
```

Author: Fixed in [6d5312c](https://github.com/eiffel-community/eiffel/pull/305/commits/6d5312c07d053524364f139d5805cf1c1c0c60d8)

* __Protocol variants__: By using the protocol it gives the best possible

>   conditions for a non-initiated event consumers to make sense of any given set of

Reviewer 2: ```suggestion
  conditions for non-initiated event consumers to make sense of any given set of
```
